pkg/rollout/rollout.go

```diff
@@ -41,22 +43,33 @@ const (
 )
 
 // New returns a new rollout manager.
-func New(client runapi.Client, metricsProvider metrics.Provider, svcRecord *ServiceRecord, strategy *config.Strategy) *Rollout {
+func New(metricsProvider metrics.Provider, svcRecord *ServiceRecord, strategy *config.Strategy) *Rollout {
 	logger := logrus.New()
 	logger.SetOutput(ioutil.Discard)
 
 	return &Rollout{
-		runClient:       client,
 		metricsProvider: metricsProvider,
 		service:         svcRecord.Service,
 		serviceName:     svcRecord.Metadata.Name,
 		project:         svcRecord.Project,
 		region:          svcRecord.Region,
 		strategy:        strategy,
-		log:             logger.WithField("project", svcRecord.Project),
+		log:             logrus.NewEntry(logrus.New()),
 	}
 }
 
+// WithClient updates the client in the rollout instance.
+func (r *Rollout) WithClient(client runapi.Client) *Rollout {
+	r.runClient = client
+	return r
+}
+
+// WithContext updates the context in the rollout instance.
+func (r *Rollout) WithContext(ctx context.Context) *Rollout {
```

Comment:
I don't get WithContext.
You can technically always pass context.TODO() or context.Background() for tests?
What's the purpose of this?

Reply:
That's true, but we also wanted to reduce the number of parameters in the constructor. 

However, it doesn't hurt to add it to `New`

Reply:
ctx is typically part of method signatures.